### DISCUSSION TITLE
CRUD Blazor Server - Editando no Forms

### DISCUSSION THREAD

**Discussion by Emerson2342** [balta-io/forum#695] [Tira Dúvidas] [answered]
Bom dia, estou fazendo um CRUD no blazor server usando o MySql como banco de dados. O back end está funcionando normal, mas o problema está na hora de editar o produto, alterar o valor e enviar de volta estes dados.
Eu preencho os dados da model com os valores vindo do banco de dados, mas quando eu tento digitar e salvar de novo num novo objeto para repassar para o endpoint, os valores não são atualizados.

@code {
    [Parameter]
    public Guid Id { get; set; }

    [SupplyParameterFromForm]
    private RequestTask model { get; set; } = new();

    private string? responseMessage;

    protected override async Task OnInitializedAsync()
    {
        var response = await Http.GetAsync($"task/task/{Id}");
        Response result = await response.Content.ReadFromJsonAsync<Response>();
       model = result.TaskList;
          responseMessage = result.Message;
    }

    private void HandleEdit()
    {

 var taskEdited = new RequestTask
        {
            Id = Id,
            Title = model.Title,
            Description = model.Description,
            StartTime = model.StartTime,
            Deadline = model.Deadline
        };
        var response = await Http.PostAsJsonAsync("task/edit", taskEdited);
        var result = await response.Content.ReadFromJsonAsync<Response>();
        responseMessage =result.Message;
        return response;
    }
}

////////aqui está meu primeiro InputText

<EditForm Model="model" OnSubmit="HandleEdit" FormName="taskEditForm">
    <DataAnnotationsValidator />
    <ValidationSummary />

    <div class="mb-1">
        <label for="Title" class="form-label">Título</label>
        <InputText class="form-control"  id="Title" @bind-Value="model.Title" />
    </div>
.
Resumindo, ao que parece, se eu preencher os dados do model com os que vem no banco de dados, ele parecem não alterar pelo input no forms, mas se eu deixar o model em branco, sem passar dados, consigo editar no forms do blazor. Está praticamente igual ao CRUD que fez no curso 2821, mas parece não funcionar. Fiz debug e realmente não "salva" os dados do input, caso o model já tiver povoado com dados.
Não é problema ao receber os dados do enpoint, pq eu coloquei os dados manualmente e mesmo assim não foi possível salvar a alteração vinda do input, conitnuou com os mesmos dados.

model.Id = Id;
        modelIn.Title = "Segunda-Feira";
        modelIn.Description = "Iniciando a semana com estágio";
        modelIn.StartTime = DateTime.UtcNow;
        modelIn.Deadline = DateTime.UtcNow.AddDays(2);
        responseMessage = "Iniciou";
.
Eu acredito que teve uma mudança no blazor para poder editar os dados. Gostaria de uma ajuda nesse aspecto. No mais, obrigado!

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @Emerson2342, como vai? Você está utilizando o RenderMode? Nas novas versões do Blazor isto é necessário:
https://balta.io/player/assistir/6263333a-258b-4077-9627-96e575e48125

**Reply by Emerson2342** [reply]
Bom dia, tudo certo. Hoje estava assistindo justamente essa aula e esse @rendermode era a solução para esse e outro problema ao tentar armazenar um token no session storage. Obrigado!
